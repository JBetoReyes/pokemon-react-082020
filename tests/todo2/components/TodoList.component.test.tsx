import React from 'react';
import { shallow } from 'enzyme';
import TodoList from '@components/hooks/Todo2/TodoList.component';
import demoTodos2 from '../../fixtures/demoTodos2';

describe('TodoList', () => {
  const handleDelete = jest.fn();
  const handleTodo = jest.fn();
  const todoListJSX = (
    <TodoList
      todos={demoTodos2}
      handleDelete={handleDelete}
      handleTodo={handleTodo}
    />
  );
  let wrapper = shallow(todoListJSX);
  beforeEach(() => {
    handleDelete.mockClear();
    handleTodo.mockClear();
    wrapper = shallow(todoListJSX);
  });
  test('Should match with the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should have two <TodoListItem /> components', () => {
    expect(wrapper.find('TodoListItem').length).toBe(demoTodos2.length);
    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(
      expect.any(Function)
    );
  });
});
