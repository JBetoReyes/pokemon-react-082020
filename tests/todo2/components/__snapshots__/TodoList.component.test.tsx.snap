// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Should match with the snapshot 1`] = `
<ol
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleTodo={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "description": "test todo 1",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleTodo={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "description": "test todo 2",
        "done": false,
        "id": 2,
      }
    }
  />
</ol>
`;
